snowflake/table_snowflake_resource_monitor.go

```diff
@@ -0,0 +1,98 @@
+      {Name: "level", Type: proto.ColumnType_STRING, Description: "Level"},
+      {Name: "frequency", Type: proto.ColumnType_STRING, Description: "Daily, Weekly, etc"},
+      {Name: "start_time", Type: proto.ColumnType_TIMESTAMP, Description: "Date and time when the monitor was started."},
+      {Name: "end_time", Type: proto.ColumnType_TIMESTAMP, Description: "Date and time when the monitor was stopped."},
+      {Name: "notify_at", Type: proto.ColumnType_STRING, Description: "Levels to which to alert"},
+      {Name: "suspend_at", Type: proto.ColumnType_STRING, Description: "Levels to which to suspend warehouse"},
+      {Name: "suspend_immediately_at", Type: proto.ColumnType_STRING, Description: "Levels to which to suspend warehouse"},
+      {Name: "created_on", Type: proto.ColumnType_TIMESTAMP, Description: "Date and time when the monitor was created."},
+      {Name: "owner", Type: proto.ColumnType_STRING, Description: "Role that owns the warehouse."},
+      {Name: "comment", Type: proto.ColumnType_STRING, Description: "Comment for the warehouse."},
+      {Name: "notify_users", Type: proto.ColumnType_STRING, Description: "Who to notify when alerting"},
+		}),
+	}
+}
+
+type ResourceMonitor struct {
+	Name              sql.NullString `json:"name" db:"name"`
+  CreditQuota       sql.NullString `json:"credit_quota" db:"credit_quota"`
+  UsedCredits       sql.NullString `json:"used_credits" db:"used_credits"`
+  RemainingCredits  sql.NullString `json:"remaining_credits" db:"remaining_credits"`
+  Level             sql.NullString `json:"level" db:"level"`
+  Frequency         sql.NullString `json:"frequency" db:"frequency"`
+  StartTime         sql.NullTime `json:"start_time" db:"start_time"`
+  EndTime           sql.NullTime `json:"end_time" db:"end_time"`
+  NotifyAt          sql.NullString `json:"notify_at" db:"notify_at"`
+  SuspendAt         sql.NullString `json:"suspend_at" db:"suspend_at"`
+  SuspendImdAt      sql.NullString `json:"suspend_immediately_at" db:"suspend_immediately_at"`
+  CreatedOn         sql.NullTime   `json:"created_on" db:"created_on"`
+	Owner             sql.NullString `json:"owner" db:"owner"`
+	Comment           sql.NullString `json:"comment" db:"comment"`
+  NotifyUsers       sql.NullString `json:"notify_users" db:"notify_users"`
+}
+
+//// LIST FUNCTION
+
+func listSnowflakeResourceMonitors(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+	logger := plugin.Logger(ctx)
+	db, err := connect(ctx, d)
+	if err != nil {
+		logger.Error("snowflake_warehouse.listSnowflakeResourceMonitors", "connnection.error", err)
+		return nil, err
+	}
+
+	rows, err := db.QueryContext(ctx, "SHOW RESOURCE MONITORS")
+	if err != nil {
+		logger.Error("snowflake_warehouse.listSnowflakeResourceMonitors", "query.error", err)
+		return nil, err
+	}
+	defer rows.Close()
+
+	dbs := []ResourceMonitor{}
```

Comment:
```suggestion
	resourceMonitors := []ResourceMonitor{}
```

We should match the variable name more closely with what we're retrieving

Reply:
Done